Controllers/EmployeeManagement/EmployeesController.cs

```diff
@@ -95,29 +97,67 @@ public ActionResult<Employee> Post([FromBody] Employee value)
         /// <param name="value">changed employee</param>
         /// <returns>one employee by id</returns>
         /// <response code="204">if the change is successful</response>
-        /// <response code="404">if bad request</response>
+        /// <response code="400">if bad request</response>
         // PUT api/<EmployeesController>/5
         [HttpPut]
         public ActionResult<Employee> Put(int id, [FromBody] Employee value)
         {
-            if (id != value.id)
+            var employee = _context.Employees.Find(id);
+
+            if (employee == null)
             {
-                return BadRequest();
+                return BadRequest("No such employee with the given id");
             }
 
+            value.id = id;
+
+            var checkResponse = Employee.CheckIfValid(value);
+
+            if (checkResponse == 1)
+                return BadRequest("Invalid body: values of Employee cannot be null");
+            else if (checkResponse == 2)
+                return BadRequest("Invalid body: invalid Employee Id");
+            else if (checkResponse == 3)
+                return BadRequest("Invalid body: invalid Employee Name");
+            else if (checkResponse == 4)
+                return BadRequest("Invalid body: invalid Employee Surname");
+            else if (checkResponse == 5)
+                return BadRequest("Invalid body: invalid Employee Email");
+            else if (checkResponse == 6)
+                return BadRequest("Invalid body: invalid Employee Username");
+            else if (checkResponse == 7)
+                return BadRequest("Invalid body: invalid Employee Password");
```

Comment:
If we make server-side validation maybe the error messages should be more explicit, for example, "No capital letters in password"?

Reply:
Yeah it would definitely be nice to have, but I think for this project this will suffice.